src/v/rpc/types.h

```diff
@@ -168,10 +169,62 @@ class streaming_context {
     std::vector<ss::semaphore_units<>> _reservations;
 };
 
+class netbuf {
+public:
+    /// \brief used to send the bytes down the wire
+    /// we re-compute the header-checksum on every call
+    ss::scattered_message<char> as_scattered() &&;
+
+    void set_status(rpc::status);
+    void set_correlation_id(uint32_t);
+    void set_compression(rpc::compression_type c);
+    void set_service_method_id(uint32_t);
+    void set_min_compression_bytes(size_t);
+    iobuf& buffer();
+
+private:
+    size_t _min_compression_bytes{1024};
+    header _hdr;
+    iobuf _out;
+};
+
+inline iobuf& netbuf::buffer() { return _out; }
+inline void netbuf::set_compression(rpc::compression_type c) {
+    vassert(
+      c >= compression_type::min && c <= compression_type::max,
+      "invalid compression type: {}",
+      int(c));
+    _hdr.compression = c;
+}
+inline void netbuf::set_status(rpc::status st) {
+    _hdr.meta = std::underlying_type_t<rpc::status>(st);
+}
+inline void netbuf::set_correlation_id(uint32_t x) { _hdr.correlation_id = x; }
+inline void netbuf::set_service_method_id(uint32_t x) { _hdr.meta = x; }
+inline void netbuf::set_min_compression_bytes(size_t min) {
+    _min_compression_bytes = min;
+}
+
+class method_probes {
+public:
+    hdr_hist& latency_hist() { return _latency_hist; }
+
+private:
+    hdr_hist _latency_hist;
```

Comment:
I think this is too expensive. that's 4KB per method, per core. Couldn't we lower the default ctor here to something more reasonable like `hdr_hist::small_hist()` or some other alias, or maybe we have a generator or default args here so that the max RPC latency is 100seconds from 1 microsecond with at the precision of 1 microsecond. Could we then lower it to be 1KB?

Reply:
good idea, should be pretty easy to do. there has been some discussion about getting rid of HDR histogram so we'll consider the size issue too.

Reply:
we can do 1ms-120s right under 2K otherwise have to choose if we want to cut off at the high end or the low end. we already have a histogram for all requests that records the lower end, and what we want is to be able to identify the high end.